akka-stream-tests/src/test/scala/akka/stream/io/OutputStreamSourceSpec.scala

```diff
@@ -68,22 +68,17 @@ class OutputStreamSourceSpec extends StreamSpec(UnboundedMailboxConfig) {
       probe.expectComplete()
     }
 
-    "block flush call until send all buffer to downstream" in assertAllStagesStopped {
-      val (outputStream, probe) = StreamConverters.asOutputStream().toMat(TestSink.probe[ByteString])(Keep.both).run
-      val s = probe.expectSubscription()
-
-      outputStream.write(bytesArray)
-      val f = Future(outputStream.flush())
-
-      expectTimeout(f, timeout)
-      probe.expectNoMsg(Zero)
-
-      s.request(1)
-      expectSuccess(f, ())
-      probe.expectNext(byteString)
-
-      outputStream.close()
-      probe.expectComplete()
+    // https://github.com/akka/akka/issues/25983
+    "not truncate the stream on close" in assertAllStagesStopped {
+      for (_ ← 1 to 10) {
+        val (outputStream, result) =
+          StreamConverters.asOutputStream()
+            .toMat(Sink.fold[ByteString, ByteString](ByteString.empty)(_ ++ _))(Keep.both)
+            .run
+        outputStream.write(bytesArray)
+        outputStream.close()
+        result.futureValue should be(ByteString(bytesArray))
+      }
```

Comment:
Would this truncate on every run or was it more random?

Reply:
This test failed pretty reliability after 2-5 iterations of the loop